CMakeLists.txt

```diff
@@ -101,8 +102,12 @@ else()
 endif()
 
 # configuring boost
+if (DEFINED ENV{BOOST_COMPILER})
+    set(Boost_COMPILER $ENV{BOOST_COMPILER})
+endif()
```

Comment:
We build project and boost with the same gcc14 compiler. But CMake detect Desbordante compiler as xgcc14 when finding boost  for some reasons: 
```
➜  desbordante-core git:(fix/macos-build-with-sdk15) ✗ ./build.sh 
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
```
Logs are correct, but after this hapenned: 
```
CMake Error at /usr/local/lib/cmake/Boost-1.86.0/BoostConfig.cmake:141 (find_package):
  Found package configuration file:

    /usr/local/lib/cmake/boost_container-1.86.0/boost_container-config.cmake

  but it set boost_container_FOUND to FALSE so package "boost_container" is
  considered to be NOT FOUND.  Reason given by package:

  No suitable build variant has been found.
  
  The following variants have been tried and rejected:
    * libboost_container-gcc14-mt-a64-1_86.dylib (gcc14, detected xgcc14, set
  Boost_COMPILER to override)

  * libboost_container-gcc14-mt-a64-1_86.a (gcc14, detected xgcc14, set
  Boost_COMPILER to override)

Call Stack (most recent call first):
  /usr/local/lib/cmake/Boost-1.86.0/BoostConfig.cmake:262 (boost_find_component)
  CMakeLists.txt:110 (find_package)
```
I think this is a bug and it needs to be reported to CMake, which I will probably do later.

Reply:
I want to clarify my question a little. I see that when [searching for a compiler](https://github.com/Kitware/CMake/blob/8bb0281896847cda8b5bf0006691dce6efe1d557/Modules/FindBoost.cmake#L956) boost uses a compiler named `xgcc`. This decision was probably not made without reason and has some motivation. And most likely this is not a bug, but was done intentionally.

I don't know what the difference is between `gcc` and `xgcc`, why this name is used in the case of boost.

And my question is rather why we, for example, do not use `xgcc` to build desbordante in the case of macOS? Then there will be no need to set the `Boost_COMPILER` variable. Why is your solution more correct?

Reply:
Actually, your link is to CMake FindBoost module and we don't use it. With new policy CMake is trying to find BoostConfig.cmake which provided in Boost. But it doesn't matter, because in the [config](https://github.com/boostorg/boost_install/blob/d77d0e94ba3d78f2079313ecb6e4907a7e8618e9/BoostDetectToolset.cmake#L88) the default gcc toolset name for macos is xgcc.